src/lib.rs

```diff
@@ -155,13 +147,13 @@ macro_rules! assert_ne {
                 if *left_val == *right_val {
                     panic!("assertion failed: `(left != right)`: {}\
                           \n\
-                          \nleft:  `{:?}`\
-                          \nright: `{:?}`\
+                          \n{}:\
+                          \n{:#?}\
                           \n\
                           \n",
                            format_args!($($arg)+),
-                           left_val,
-                           right_val)
+                           $crate::Style::new().bold().paint("Both sides"),
```

Comment:
Beware, `assert_ne!` is a tricky beast! :-)

Just because `PartialEq` says that `left_val` and `right_val` are equivalent, doesn't mean they are structurally the same or have the same `Display` result.

E.g. [playpen: `assert_ne!(-0., 0.)`](https://gist.github.com/3b4bcfdf933188021a8024bd78b2846a).

If we want to collapse the representation to `both_sides: {left_val}` we must first check if both have the same `Display` representation!

PS: As to your original question - I'm not a fan of underline. I'd prefer bold.